File: packages/http-client-csharp/generator/TestProjects/Spector.Tests/Http/Payload/Head/HeadTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Threading.Tasks;
using NUnit.Framework;
using Payload.Head;

namespace TestProjects.Spector.Tests.Http.Payload.Head
{
    public class HeadTests : SpectorTestBase
    {
        [SpectorTest]
        public Task ContentTypeHeaderInResponse() => Test(async (host) =>
        {
            var response = await new HeadClient(host, null).ContentTypeHeaderInResponseAsync();
            var rawResponse = response.GetRawResponse();
            Assert.AreEqual(200, rawResponse.Status);
            Assert.IsTrue(rawResponse.Headers.TryGetValue("Content-Type", out string? contentType));
            Assert.AreEqual("text/plain; charset=utf-8", contentType);
            Assert.IsTrue(rawResponse.Headers.TryGetValue("x-ms-meta", out string? metadata));
            Assert.AreEqual("hello", metadata);
        });
    }
}

File: packages/http-client-java/generator/http-client-generator-clientcore-test/src/test/java/payload/head/HeadTests.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package payload.head;

import io.clientcore.core.http.models.HttpHeaderName;
import io.clientcore.core.http.models.RequestContext;
import io.clientcore.core.http.models.Response;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class HeadTests {

    private final HeadClient client = new HeadClientBuilder().buildClient();

    @Test
    public void testResponseHeaders() {
        Response<Void> response = client.contentTypeHeaderInResponseWithResponse(RequestContext.none());
        Assertions.assertEquals("text/plain; charset=utf-8",
            response.getHeaders().getValue(HttpHeaderName.CONTENT_TYPE));
        Assertions.assertEquals("hello", response.getHeaders().getValue(HttpHeaderName.fromString("x-ms-meta")));
    }
}
